### PR TITLE
Improve dpnp.log() implementation

### PR DESCRIPTION
**Pull request by antonwolfy**
The new logic assumes to call `ln` from pybind11 extension of OneMKL VM if possible or fully rely on dpctl.tensor implementation for `log` otherwise.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [x] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/backend/extensions/vm/common.hpp [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+    int dst_typeid = array_types.typenum_to_lookup_id(dst_typenum);
+
+    if (src_typeid != dst_typeid) {
+        throw py::value_error("Input and output arrays have different types.");
+    }
+
+    // check that queues are compatible
+    if (!dpctl::utils::queues_are_compatible(exec_q, {src, dst})) {
+        throw py::value_error(
+            "Execution queue is not compatible with allocation queues");
```
Comment (vlad-perevezentsev) [inline dpnp/backend/extensions/vm/common.hpp:93]:
```suggestion
            "Execution queue is not compatible with allocation queues.");
```

File: dpnp/dpnp_algo/dpnp_elementwise_common.py [hunk 2]
```diff
@@ -374,6 +446,54 @@ def dpnp_less_equal(x1, x2, out=None, order="K"):
     return dpnp_array._create_from_usm_ndarray(res_usm)
 
 
+_log_docstring = """
+log(x, out=None, order='K')
+Computes the natural logarithm element-wise.
+Args:
+    x (usm_ndarray):
+        Input array, expected to have numeric data type.
+    out (usm_ndarray):
```
Comment (vlad-perevezentsev) [inline dpnp/dpnp_algo/dpnp_elementwise_common.py:455]:
```suggestion
    out ({None, usm_ndarray}, optional):
```

File: tests/helper.py [hunk 3]
```diff
@@ -90,6 +90,15 @@ def is_win_platform():
     return platform.startswith("win")
 
 
+def has_support_aspect16(device=None):
```
Comment (vlad-perevezentsev) [inline tests/helper.py:93]: This function is defined below (line 102)

File: dpnp/dpnp_algo/dpnp_elementwise_common.py (hunk 2 above)
Comment (antonwolfy) [inline dpnp/dpnp_algo/dpnp_elementwise_common.py:455]:
```suggestion
    x (dpnp.ndarray):
        Input array, expected to have numeric data type.
    out ({None, dpnp.ndarray}, optional):
```

File: dpnp/dpnp_algo/dpnp_elementwise_common.py [hunk 4]
```diff
@@ -374,6 +446,54 @@ def dpnp_less_equal(x1, x2, out=None, order="K"):
...
+    x (dpnp.ndarray):
+        Input array, expected to have numeric data type.
+    out ({None, dpnp.ndarray}, optional):
+        Output array to populate. Array must have the correct
+        shape and the expected data type.
+    order ("C","F","A","K", optional): memory layout of the new
+        output array, if parameter `out` is `None`.
+        Default: "K".
+Return:
+    usm_ndarray:
```
Comment (antonwolfy) [inline dpnp/dpnp_algo/dpnp_elementwise_common.py:462]:
```suggestion
    dpnp.ndarray:
```

**Review by vlad-perevezentsev — APPROVED** [review]
LGTM
thanks @antonwolfy !

**Pull request merged**